tests/shared/src/test/scala-2.13/scala/meta/tests/parsers/dotty/Scala3PositionSuite.scala

```diff
@@ -661,4 +866,68 @@ class Scala3PositionSuite extends BasePositionSuite(dialects.Scala3) {
        |""".stripMargin
   )
 
+  checkPositions[Stat](
+    """|val (x, y) = (foo, bar) match
+       |  case (false, false) => (baz, qux) // c1""".stripMargin,
+    """|Pat.Tuple (x, y)
+       |Term.Match (foo, bar) match
+       |  case (false, false) => (baz, qux) // c1
+       |Term.Tuple (foo, bar)
+       |Case case (false, false) => (baz, qux)
+       |Pat.Tuple (false, false)
+       |Term.Tuple (baz, qux)
+       |""".stripMargin
+  )
+
+  checkPositions[Stat](
+    """|val (x, y) = (foo, bar) match
+       |  case (false, false) => (baz, qux) // c1
+       |""".stripMargin,
+    """|Pat.Tuple (x, y)
+       |Term.Match (foo, bar) match
+       |  case (false, false) => (baz, qux) // c1
+       |Term.Tuple (foo, bar)
+       |Case case (false, false) => (baz, qux)
+       |Pat.Tuple (false, false)
+       |Term.Tuple (baz, qux)
+       |""".stripMargin
+  )
+
+  checkPositions[Stat](
+    """|val (x, y) = (foo, bar) match
+       |  case (true, false) => (baz, qux) // c1
+       |  case (false, true) => (baz, qux) // c2
+       |""".stripMargin,
+    """|Pat.Tuple (x, y)
+       |Term.Match (foo, bar) match
+       |  case (true, false) => (baz, qux) // c1
+       |  case (false, true) => (baz, qux) // c2
+       |Term.Tuple (foo, bar)
+       |Case case (true, false) => (baz, qux) // c1
+       |Pat.Tuple (true, false)
+       |Term.Tuple (baz, qux)
+       |Case case (false, true) => (baz, qux)
+       |Pat.Tuple (false, true)
+       |Term.Tuple (baz, qux)
+       |""".stripMargin
+  )
+
+  checkPositions[Stat](
+    """|val (x, y) = (foo, bar) match
+       |  case (true, false) => (baz, qux) // c1
+       |  case (false, true) => (baz, qux) // c2""".stripMargin,
+    """|Pat.Tuple (x, y)
+       |Term.Match (foo, bar) match
+       |  case (true, false) => (baz, qux) // c1
+       |  case (false, true) => (baz, qux) // c2
+       |Term.Tuple (foo, bar)
+       |Case case (true, false) => (baz, qux) // c1
+       |Pat.Tuple (true, false)
+       |Term.Tuple (baz, qux)
+       |Case case (false, true) => (baz, qux)
```

Comment:
Shouldn't the second `case` include `// c2`?

Reply:
good point. how do i do it? how do i find the code which sets the end position for case? perhaps it didn't use autopos there.

Reply:
i think i figured it out. will add a separate commit fixing this. but first, please review and approve #2553.